src/pages/index.js

```diff
@@ -84,24 +90,148 @@ function handleImageClick(data) {
+      editProfileModal.setLoading(false);
+    })
+    .catch((err) => {
+      console.error("Error updating user info", err);
+      alert(err);
+    });
   editProfileModal.close();
 }
 function handleAddCardFormSubmit(formValues) {
   const name = formValues.title;
   const link = formValues.link;
+  // Make API request to upload card
+  addCardModal.setLoading(true);
+  api
+    .uploadCard({ name, link })
+    .then((cardData) => {
+      const card = createCard(cardData);
+      addCardModal.setLoading(false);
 
-  const card = createCard({ name, link });
-  cardList.addItem(card);
-  console.log(formValues);
-  cardAddForm.reset();
-  addCardModal.close();
+      cardList.addItem(card);
+      addCardModal.close();
+      cardAddForm.reset();
+    })
+    .catch((error) => {
+      console.error(error);
+    });
 }
 
 function createCard(data) {
-  const card = new Card(data, "#card-template", handleImageClick);
+  const card = new Card(
+    data,
+    "#card-template",
+    handleImageClick,
+    handleDeleteCardSubmit,
+    likeCard,
+    unlikeCard
+  );
   return card.generateCard();
 }
 
-//Event Listeners
+api
+  .getInitialCards()
+  .then((res) => {
+    console.log(res);
+    cardList.renderItems(res);
+  })
+
+  .catch((err) => alert(err));
+
+api
+  .getUserInfo()
+  .then((res) => {
+    userInfo.setUserInfo(res);
+    userInfo.updateProfileImage(res);
+  })
+  .catch((err) => alert(err));
+
+//Avatar
+const profileImageForm = document.querySelector("#edit-avatar-form");
+const profileFormValidator = new FormValidator(
+  validationConfig,
+  profileImageForm
+);
+profileFormValidator.enableValidation();
+
+function handleImageProfileEditSubmit(data) {
+  newProfileImageModal.setLoading(true);
+
+  api
+    .setUserAvatar(data.link)
+    .then((res) => {
+      userInfo.updateProfileImage(res);
+      newProfileImageModal.close();
+      profileImageCover.reset();
+      newProfileImageModal.setLoading(false);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
+const profileImageCover = document.querySelector(".profile__edit-image");
+profileImageCover.addEventListener("click", () => {
+  newProfileImageModal.open();
+});
+
+const newProfileImageModal = new PopupWithForm({
+  popupSelector: "#edit-avatar-modal",
+  handleFormSubmit: handleImageProfileEditSubmit,
+});
+newProfileImageModal.setEventListeners();
+
+// confirmation
+
+function handleDeleteCardSubmit(card) {
+  confirmModal.setSubmitAction(() => {
+    api
+      .deleteCard(card._id)
+      .then((res) => {
+        console.log(res);
+        card.remove();
+        confirmModal.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  });
+  confirmModal.open(card);
+}
+
+const confirmModal = new PopupWithConfirmation({
+  popupSelector: "#confirmation-modal",
+});
+confirmModal.setEventListeners();
+
+function likeCard(card) {
+  api
+    .likeCard(card._id, card._isLiked)
```

Comment:
`_isLiked`

Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.